tests/expectations/metrics/test_core.py

```diff
@@ -1703,6 +1703,75 @@ def test_map_column_pairs_in_set_metric_pd():
     )
 
 
+def test_map_column_pairs_in_set_metric_sa(sa):
+    engine = build_sa_engine(
+        pd.DataFrame({"a": [10, 3, 4, None, 3, None], "b": [1, 2, 3, None, 3, 5]}), sa
+    )
+
+    metrics: dict = {}
+
+    table_columns_metric: MetricConfiguration
+    results: dict
+
+    table_columns_metric, results = get_table_columns_metric(engine=engine)
+    metrics.update(results)
+
+    condition_metric = MetricConfiguration(
+        metric_name="column_pair_values.in_set.condition",
+        metric_domain_kwargs={
+            "column_A": "a",
+            "column_B": "b",
+            "ignore_row_if": "either_value_is_missing",
+        },
+        metric_value_kwargs={
+            "value_pairs_set": [(2, 1), (3, 2), (4, 3), (3, 3)],
+            "result_format": {
+                "result_format": "SUMMARY",
+                "partial_unexpected_count": 6,
+            },
+        },
+        metric_dependencies={
+            "table.columns": table_columns_metric,
+        },
+    )
+    results = engine.resolve_metrics(
+        metrics_to_resolve=(condition_metric,),
+        metrics=metrics,
+    )
+    metrics.update(results)
+
+    assert isinstance(metrics[condition_metric.id][0], sa.sql.elements.AsBoolean)
```

Comment:
Do we want to add an assertion message here? This is an invariant that should never fail?

Reply:
This shouldn't fail unless the metric returns the wrong type. Part of the reason for the `condition_metric` is to use it below when getting the `unexpected_values`.

Reply:
@cdkini this should only fail if the metric is altered to return some other type. The main reason this `condition_metric` is computed is to be used in the `unexpected_values_metric` in the lines below.